cdci_data_analysis/analysis/instrument.py

```diff
@@ -138,31 +138,32 @@ def set_pars_from_dic(self, par_dic, verbose=False):
         updated_par_dic = par_dic.copy()
         
         for par in param_list:
-            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
-            # or they don't have to set (eg scw_list)
-
             self.logger.info("before normalizing, set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
                              par, par.name, par.value, par_dic.get(par.name, None))
 
+            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
+            # or they don't have to set (eg scw_list)
             if par.name is not None and par.name not in params_not_to_be_included:
-                # set the value for par in to a default format,
+                # set the value for par to a default format,
                 # or to a default value if this is not included within the request
-                
                 updated_par_dic[par.name] = par.set_value_from_form(par_dic, verbose=verbose)
 
-                if par.default_units is not None:
+                if par.units_name is not None and par.default_units is not None:
                     updated_par_dic[par.units_name] = par.default_units
 
+                if par.units_name is not None and par.default_units is None:
+                    raise RequestNotUnderstood("Error when setting the parameter %s: "
+                                               "default unit not specified" % par.name)
+                if par.units_name is None and par.default_units is not None:
```

Comment:
would you consider an action also in case both fields were not available?

Reply:
> would you consider an action also in case both fields were not available?

A warning.

Reply:
By the way, it's not `RequestNotUnderstood` it's `InternalError`: this can only happen if constructor was called incorrectly, and this is done in plugins, in  the code. It's not user error.